compiler/rustc_hir_typeck/src/closure.rs

```diff
@@ -52,6 +52,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     ) -> Ty<'tcx> {
         trace!("decl = {:#?}", closure.fn_decl);
 
+        deny_non_region_late_bound(self.tcx, closure.bound_generic_params, "closure");
```

Comment:
All this checks should probably be in `rustc_hir_analysis::collect::lifetimes`.

Reply:
I attempted to implement something like that. If binder doesn't allow late-bound ty or consts, then it'll resolve to `ResolvedArg::Error` instead, so we never actually see these late-bound type or const variables showing up where they shouldn't be.

Not sure if this is what you were looking for.